nifi/nifi-nar-bundles/nifi-kite-bundle/nifi-kite-processors/src/main/java/org/kitesdk/data/spi/filesystem/CSVFileReaderFixed.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2013 Cloudera Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kitesdk.data.spi.filesystem;
+
+import au.com.bytecode.opencsv.CSVReader;
+import java.io.InputStream;
+import com.google.common.collect.Lists;
+import java.util.List;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.kitesdk.data.DatasetDescriptor;
+import org.kitesdk.data.DatasetIOException;
+import org.kitesdk.data.spi.AbstractDatasetReader;
+import org.kitesdk.data.spi.DescriptorUtil;
+import org.kitesdk.data.spi.EntityAccessor;
+import org.kitesdk.data.spi.ReaderWriterState;
+import com.google.common.base.Preconditions;
+import org.apache.avro.Schema;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.kitesdk.data.spi.filesystem.CSVProperties;
+import org.kitesdk.data.spi.filesystem.CSVUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.NoSuchElementException;
+
+import static org.kitesdk.data.spi.filesystem.FileSystemProperties.REUSE_RECORDS;
+
+/**
+ * This is a temporary addition. The version in 0.18.0 throws a NPE when the
+ * InputStream constructor is used.
+ */
+public class CSVFileReaderFixed<E> extends AbstractDatasetReader<E> {
```

Comment:
What happens if someone is using this version of the reader but drops in a later version of Kite?

Can we make this a non-public class that isn't in the package space of Kite? Could we make it a smaller class that inherited from the Kite class and corrected just the specific behavior we're trying to avoid?